File: android/ui/src/main/kotlin/com/gemwallet/android/ui/components/empty/EmptyContentType.kt
package com.gemwallet.android.ui.components.empty

import uniffi.gemstone.GemEmptyStateAction
import uniffi.gemstone.GemEmptyStateKind

class EmptyContentType(val kind: GemEmptyStateKind, val symbol: String = "", val isViewOnly: Boolean = false, actions: Map<GemEmptyStateAction, (() -> Unit)?> = emptyMap()) {
    val actions: Map<GemEmptyStateAction, () -> Unit> = actions.mapNotNull { (action, onClick) -> onClick?.let { action to it } }.toMap()
}

File: ios/Packages/PrimitivesComponents/Sources/Types/EmptyContentType.swift
// Copyright (c). Gem Wallet. All rights reserved.

public import enum Gemstone.GemEmptyStateAction
public import enum Gemstone.GemEmptyStateKind
import Foundation

public struct EmptyContentType {
    let kind: GemEmptyStateKind
    let symbol: String
    let isViewOnly: Bool
    let actions: [GemEmptyStateAction: () -> Void]

    public init(
        _ kind: GemEmptyStateKind,
        symbol: String = "",
        isViewOnly: Bool = false,
        actions: [GemEmptyStateAction: (() -> Void)?] = [:],
    ) {
        self.kind = kind
        self.symbol = symbol
        self.isViewOnly = isViewOnly
        self.actions = actions.compactMapValues { $0 }
    }
}
